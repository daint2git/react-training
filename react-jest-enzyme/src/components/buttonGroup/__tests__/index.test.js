import ButtonGroup from '..'

describe('ButtonGroup', () => {
  const list = [1, 2]
  test('using shallow', () => {
    const shallowed = shallow(<ButtonGroup list={list} />)

    expect(shallowed).toMatchSnapshot()
  })

  test('using mount', () => {
    const mounted = mount(<ButtonGroup list={list} />)

    expect(mounted).toMatchSnapshot()
    mounted.unmount()
  })

  test('using render', () => {
    const rendered = render(<ButtonGroup list={list} />)

    expect(rendered).toMatchSnapshot()
  })

  test('using react-test-renderer', () => {
    const tree = renderer.create(<ButtonGroup list={list} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('button clicked', () => {
    const callBack = jest.fn()
    const list = ['btn1', 'btn2']
    const wrapper = shallow(<ButtonGroup list={list} onClick={callBack} />)

    const btns = wrapper.find('Button')
    expect(btns.length).toBe(2)

    const btn1 = btns.first()
    btn1.simulate('click')
    expect(callBack).toHaveBeenCalled()
    expect(callBack).toHaveBeenCalledTimes(1)
    expect(callBack).toHaveBeenCalledWith('btn1')

    const btn2 = btns.at(1)
    btn2.simulate('click')
    expect(callBack).toHaveBeenCalled()
    expect(callBack).toHaveBeenCalledTimes(2)
    expect(callBack).toHaveBeenCalledWith('btn2')
  })

  test('props', () => {
    const callBack = jest.fn()
    const wrapper = mount(<ButtonGroup list={list} onClick={callBack} />)
  })
})
