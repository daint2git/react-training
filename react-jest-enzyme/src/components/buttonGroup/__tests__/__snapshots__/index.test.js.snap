// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup using mount 1`] = `
<styled.div>
  <div
    className="sc-bwzfXH jIEWCi"
  >
    <styled.button
      className="class"
      onClick={[Function]}
      type="button"
    >
      <button
        className="class sc-bdVaJa ihDGdR"
        onClick={[Function]}
        type="button"
      >
        <strong>
          1
        </strong>
      </button>
    </styled.button>
    <styled.button
      className="class"
      onClick={[Function]}
      type="button"
    >
      <button
        className="class sc-bdVaJa ihDGdR"
        onClick={[Function]}
        type="button"
      >
        <strong>
          2
        </strong>
      </button>
    </styled.button>
  </div>
</styled.div>
`;

exports[`ButtonGroup using react-test-renderer 1`] = `
<div
  className="sc-bwzfXH jIEWCi"
>
  <button
    className="class sc-bdVaJa ihDGdR"
    onClick={[Function]}
    type="button"
  >
    <strong>
      1
    </strong>
  </button>
  <button
    className="class sc-bdVaJa ihDGdR"
    onClick={[Function]}
    type="button"
  >
    <strong>
      2
    </strong>
  </button>
</div>
`;

exports[`ButtonGroup using render 1`] = `
<div
  class="sc-bwzfXH jIEWCi"
>
  <button
    class="class sc-bdVaJa ihDGdR"
    type="button"
  >
    <strong>
      1
    </strong>
  </button>
  <button
    class="class sc-bdVaJa ihDGdR"
    type="button"
  >
    <strong>
      2
    </strong>
  </button>
</div>
`;

exports[`ButtonGroup using shallow 1`] = `
<styled.div>
  <Button
    className="class"
    key="1"
    onClick={[Function]}
  >
    <strong>
      1
    </strong>
  </Button>
  <Button
    className="class"
    key="2"
    onClick={[Function]}
  >
    <strong>
      2
    </strong>
  </Button>
</styled.div>
`;
